src/+otp/+kuramotosivashinsky/+presets/Canonical.m

```diff
@@ -0,0 +1,49 @@
+% The Kuramoto-Sivashinky equation is a chaotic problem.
+%
+% In this particular discretization, we are applying a spectral
+% method, therefore the boundary conditions will be chosen to be as cyclic
+% on the domain [0, L]. Note that this is different from another typical
+% domain of [-L, L]. The larger the L, the more interesting the problem is
+% but the more points are required to do a good discretization. The current
+% canonical implementation with the size, L, and  is used in
+%
+% Kassam, Aly-Khan, and Lloyd N. Trefethen. 
+% "Fourth-order time-stepping for stiff PDEs." 
+% SIAM Journal on Scientific Computing 26, no. 4 (2005): 1214-1233.
+%
+
+classdef Canonical < otp.kuramotosivashinsky.KuramotoSivashinskyProblem
+    
+    methods
+        function obj = Canonical(varargin)
+            
+            p = inputParser;
+            addParameter(p, 'Size', 128);
+            addParameter(p, 'L', 32*pi);
+
+            parse(p, varargin{:});
+            
+            s = p.Results;
+            
+            N = s.Size;
+            L = s.L;
+            
+            params.L = L;
+            
+            h = L/N;
+            
+            % exclude the left boundary point as it is identical to the
+            % right boundary point
+            x = linspace(h, L, N).';
+            
+            u0 = cos(x/16).*(1+sin(x/16));
```

Comment:
Replace hardcoded 16 so i.c. is periodic for different L.  It might be easier to have x in the range 0 to 2 pi.

Reply:
If I have x in 0 to 2 pi, then I would have to calculate h differently. I will change u0, to be a function of L and use cospi and sinpi. I think that is simpler.